embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionMessageCollator.kt

```diff
@@ -147,10 +148,22 @@
         )
     }
 
+    private inline fun <R> captureSafely(result: () -> R): R? {
```

Comment:
I wonder if we can stick this in a location where it can be used in the various places we have to do this. And it might be good if we have have an indication of what failed in the internal error so we know what exactly is going wrong.

This could be done in a separate PR BTW